spec/draft-barnes-cfrg-4q.md

```diff
@@ -268,53 +266,50 @@ computed, e.g., as mask(c) = 0 - c.
-
-We use the following 3 representations of a point (x, y) on the
-curve. All representations use X, Y, Z satisfying x = X/Z, y = Y/Z. The
-point at infinity is (0,1,1). These representations differ in auxiliary
-data used to speed some operations. By omitting their computation when
-they are not needed we save operations. In three of these representations
-the additional coordinate T=XY/Z is used to define them.
-
-Point representation R1 is given by (X,Y,Z,Ta,Tb), where T=Ta*Tb. Representation R2 is
-(N, D, E, F) = (X+Y,Y-X,2Z,2dT). Representation R3 is (N, D, Z, T) = (X+Y,Y-X,Z,T), and  
-representation R4 is (X,Y,Z).  A point doubling takes an R4 point and produces an R1 point.  
-There are two kinds of addition: ADD_core eats an R2 and an R3 point and produces an
-R1 point, and ADD eats an R1 and an R2 point, by converting the R1
-point into an R3 point and then proceeding. Exposing these two
-operations and the multiple representations helps save time in
-precomputing tables and in using the tables effectively. The conversions
-between point formats are obvious.
-
-These operations have the following explicit formulas developed by
+### Alternative Point Representations and Addition Laws
+
+We use projective coordinates based on extended twisted Edwards coordinates [TwistedRevisited]:
+the projective tuple (X, Y, Z, T) with Z != 0 and T = X * Y/Z corresponds to a point (x, y)  
+satisfying x = X/Z and y = Y/Z. The point at infinity is (0,1,1). The following variants are used in 
+the optimized scalar multiplication algorithm in order to save computations: point representation 
+R1 is given by (X,Y,Z,Ta,Tb), where T=Ta*Tb; representation R2 is (N, D, E, F) = (X+Y,Y-X,2Z,2dT); 
+representation R3 is (N, D, Z, T) = (X+Y,Y-X,Z,T); and representation R4 is (X,Y,Z).  
+A point doubling (DBL) takes an R4 point and produces an R1 point. For addition, we first define ADD_core that takes 
+an R2 and R3 point and produces an R1 point. Then, a point addition (ADD), which takes an R1 and R2 point as inputs, 
+first converts the R1 point to R3 and then executes ADD_core. Exposing these operations and the multiple representations 
+helps save time during table precomputation and the actual scalar multiplication. Conversion
+between point representations is straightforward.
+
+The point operations above have the following explicit formulas developed by
 many people over the years ([EFD],[TwistedRevisted]). We present
 the operations as functions in pseudocode.
 
+[[NOTE: IMO we should directly credit the authors for their formulas. This includes Bernstein et al., 2008
+and Hisil et al., 2008, as far as I know.]]
+
 Doubling is computed as follows
 
 ~~~~
 DBL(X1, Y1, Z1):
   A = X1^2
   B = Y1^2
   C = 2*Z1^2
-  D = -1*A
-  E = (X1+Y1)^2-A-B
-  G = D+B
-  F = G-C
-  H = D-B
-  X3 = E*F
-  Y3 = G*H
+  D = A+B
+  E = (X1+Y1)^2-D
+  F = B-A
+  G = C-F
+  X3 = E*G
+  Y3 = D*F
   Ta3 = E
-  Tb3 = H
+  Tb3 = D
   Z3 = F*G
```

Comment:
Is there a source for this doubling formula? It doesn't match what is in the EFD for twisted Edwards curves with extended coordinates.